drivers/ethernet/eth_stm32_hal.c

```diff
@@ -222,8 +225,13 @@ static inline void disable_mcast_filter(ETH_HandleTypeDef *heth)
 	ETH_MACFilterConfigTypeDef MACFilterConf;
 
 	HAL_ETH_GetMACFilterConfig(heth, &MACFilterConf);
+#if defined(ETH_STM32_MULTICAST_FILTER)
```

Comment:
```suggestion
#if defined(CONFIG_ETH_STM32_MULTICAST_FILTER)
```

Reply:
Thanks for spotting that! I've fixed it in my latest commit.